docs/build_pipeline_controller.md

```diff
@@ -0,0 +1,36 @@
+<div align="center"><h1>BuildPipeline Controller</h1></div>
+
+```mermaid
+
+%%{init: {'theme':'forest'}}%%
+flowchart TD
+%% Defining the styles
+  classDef Red fill:#FF9999;
+  classDef Amber fill:#FFDEAD;
+  classDef Green fill:#BDFFA4;
+ 
+  %% Node definitions
+predicate((PREDICATE: <br> filter events related to <br> PipelineRun <br> that are signed and <br> have succeeded))
+get_pipeline_run{Get the pipeline run<br>from the cluster}
+get_component{Get the component<br>associated with the pipeline run}
+get_application{Get the application<br>associated with the component}
```

Comment:
I think that getting application and component can probably be rolled into the same node.